### ISSUE TITLE
Tooltip: should be hoverable and dismissible for accessibility

### ISSUE THREAD

**Issue by miedrl** [primefaces/primefaces#7580]
**Describe the defect**
The components tooltip and overlayPanel do not fullfil the WCAG 2.1 [success criterion 1.4.13](https://www.w3.org/TR/WCAG21/#content-on-hover-or-focus) for accessibility. The points "dismissible" and "hoverable" of the criterion are not fulfilled as the tooltip/overlay panel disappear when hovering over them and are not dismissible via e.g. ESC.

**Reproducer**
Use this reproducer [reproducer.zip](https://github.com/primefaces/primefaces/files/6813986/reproducer.zip), run **mvn clean jetty:run** and then go to http://localhost:8080/primefaces-test/test.xhtml

**Environment:**
 - PF Version: _10.0_ with Rain Theme

**To Reproduce**
1. Mouse Hover over the input text in the reproducer, a tooltip or overlay panel will appear.
2. Try to hover over the tooltip/overlay panel.
4. See error: The tooltip/overlay panel disappears when trying to hover over it.

**Expected behavior**

**Comment by melloware** [comment]
@miedrl you don't list what is your expected behavior?  I am not understanding exactly what is wrong?

When you hover over it and it dissappears that is controlled by the `hideEvent` option see: https://primefaces.github.io/primefaces/10_0_0/#/components/tooltip

The default is `hideEvent="mouseout"` but you can change that to be another JS Event to hide it.

And for OverlayPanel the defaults are 

https://primefaces.github.io/primefaces/10_0_0/#/components/overlaypanel?id=overlaypanel
showEvent | click | String | Event on target to show the panel.
hideEvent | click | String | Event on target to hide the panel.
dismissable | true | Boolean | When set true, clicking outside of the panel hides the overlay.

**Comment by miedrl** [comment]
@melloware 
I'm sorry, i forgot to add the expected behaviour but it is also mentioned in WCAG 2.0 success criterion 1.4.13 which I linked to. This criterion (among others) has to be fulfilled so that a website or web application can be called accessible.

Additional contents such as PrimeFaces' tooltips and overlay panels need to be dismissible and hoverable as follows:
    
**dismissible:**
> A mechanism is available to dismiss the additional content without moving pointer hover or keyboard focus, unless the
>  additional content communicates an input error or does not obscure or replace other content.

--> This means that there has to be a way to close a tooltip/overlay panel without hovering out e.g. via pressing ESC

**hoverable:**
> If pointer hover can trigger the additional content, then the pointer can be moved over the additional content without the
   additional content disappearing."

--> This means if one hovers over e.g. an input text and a tooltip appears, one must be able to move the hover 
      over the tooltip (away from the input text) without the tooltip closing. Same goes with overlay panels.

**Comment by melloware** [comment]
Dismissable makes sense to add ESC to close it.

However for hoverable see my post above.  This is controlled with "showEvent" and "hideEvent" so there is nothing we can FORCE it to stay open if entered if someone had say `"hideEvent="mouseover"` right?  Without taking that flexibility away from the component?

**Comment by miedrl** [comment]
@melloware 
Is there a way to expand the mouseover and mouseout events of the button so that they also cover the tooltip/overlay panel? This way, the tooltip would only close if the mouse is outside the button AND outside the tooltip.

Another problem appears with overlay panel (collision is set to "fit") if the overlay panel is very large and the screen width is very small. The content looks like this:
![grafik](https://user-images.githubusercontent.com/77431692/125626380-3d68bb5e-293c-415d-870e-7c9b40aec405.png)

If I hover over the button, an overlay panel appears whicht covers the whole screen. The problem is that the overlay panel flickers (keeps on appearing and disappearing) because the mouseout event is triggered when the panel appears. The panel then disappears and the mouse is on the button again triggering the mousover event and so on.
![grafik](https://user-images.githubusercontent.com/77431692/125626907-311519a9-4ef0-485a-834d-21023b507aa9.png)

This problem would also disappear if the mouse events also cover the area of the tooltip/overlay panel.

**Comment by melloware** [comment]
Not sure i would have to research.

**Comment by miedrl** [comment]
@melloware Has a solution for this problem been found?

**Comment by melloware** [comment]
I have not had a chance to look into this.  If you have time to research a solution any help would be appreciated.

**Comment by melloware** [comment]
@miedrl shoot me a private message about this issue when you get a chance.

**Comment by miedrl** [comment]
@melloware Sorry but I can not provide a solution.

**Comment by melloware** [comment]
Are you or your company interested in sponsoring this fix financially?

**Comment by miedrl** [comment]
Sorry but no, we won't be sponsoring this fix.

**Comment by melloware** [comment]
OK I have a fix for tooltip that adds `autoHide` property you can see in Action in PrimeReact showcase here: https://primefaces.org/primereact/tooltip/

Please open a separate issue about OverlayPanel so it can be tracked separately.

**Comment by miedrl** [comment]
@melloware Like that? https://github.com/primefaces/primefaces/issues/8546
